controllers/etcd_controller.go

```diff
@@ -1124,23 +1138,63 @@ func (r *EtcdReconciler) waitUntilStatefulSetReady(sts *appsv1.StatefulSet) (*ap
 		}
 		return gardenerretry.Ok()
 	})
+	if err != nil {
+		messages, err2 := r.fetchPVCEventsFor(ctx, ss)
+		if err2 != nil {
+			logger.Error(err2)
+			// don't expose this error since fetching events is a best effort
+			// and shouldn't be confused with the actual error
+			return ss, err
+		}
+		if messages != "" {
+			return ss, fmt.Errorf("%w\n\n%s", err, messages)
+		}
+	}
+
 	return ss, err
 }
 
-func (r *EtcdReconciler) removeOperationAnnotation(etcd *druidv1alpha1.Etcd) error {
+func (r *EtcdReconciler) fetchPVCEventsFor(ctx context.Context, ss *appsv1.StatefulSet) (string, error) {
+	pvcs := &corev1.PersistentVolumeClaimList{}
+	if err := r.List(ctx, pvcs, client.InNamespace(ss.GetNamespace())); err != nil {
+		return "", err
+	}
+
+	var (
+		pvcMessages  string
+		volumeClaims = ss.Spec.VolumeClaimTemplates
+	)
+	for _, volumeClaim := range volumeClaims {
+		for _, pvc := range pvcs.Items {
+			if pvc.Status.Phase != corev1.ClaimBound && !strings.HasPrefix(pvc.GetName(), fmt.Sprintf("%s-%s", volumeClaim.Name, ss.Name)) {
```

Comment:
If you're using label selector above, you won't need to have the PVC name check condition here.